### PR TITLE
Update audio related logic

### PR DESCRIPTION
**Pull request by ihhub**
- add an environmental sound for Freeman's Foundry
- update sounds and music before a hero making an action
- update comments for unused sounds to track them in the future
- code cleanup for Music

### CODE REVIEW

File: src/fheroes2/heroes/heroes_action.cpp [hunk 1]
```diff
@@ -246,17 +246,30 @@ static void WhirlpoolTroopLoseEffect( Heroes & hero )
...
-    if ( GetKingdom().isControlAI() )
+    if ( GetKingdom().isControlAI() ) {
+        // Restore the original music after the action is completed.
+        const Game::MusicRestorer musicRestorer;
         return AI::HeroesAction( *this, tileIndex );
+    }
+
+    // Update environment sounds and music before doing the action. Interface::Basic::SetFocus() function is responsible for update them after the action.
+    const int32_t heroPosIndex = GetIndex();
+    if ( Game::UpdateSoundsOnFocusUpdate() && heroPosIndex >= 0 ) {
```
Comment (oleg-derevenetz) [inline src/fheroes2/heroes/heroes_action.cpp:257]: Is there any legitimate chance that `heroPosIndex` will be negative here? If not, maybe replace this with an assertion?

**Review by oleg-derevenetz — COMMENTED** [review]
Hi @ihhub I just left one question, could you please take a look when you have time?

File: src/fheroes2/heroes/heroes_action.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/heroes/heroes_action.cpp]: Done.

**Review by oleg-derevenetz — APPROVED** [review]

File: src/fheroes2/heroes/heroes_action.cpp [hunk 2]
```diff
@@ -246,17 +246,31 @@ static void WhirlpoolTroopLoseEffect( Heroes & hero )
 // action to next cell
 void Heroes::Action( int tileIndex, bool isDestination )
 {
-    // restore the original music after the action is completed
-    const Game::MusicRestorer musicRestorer;
-
-    if ( GetKingdom().isControlAI() )
+    if ( GetKingdom().isControlAI() ) {
+        // Restore the original music after the action is completed.
+        const Game::MusicRestorer musicRestorer;
```
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/heroes/heroes_action.cpp:251]: @ihhub Yes, after a battle it may come in handy.

**Pull request merged**